cpp/src/svm/ws_util.cu

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ml_utils.h"
+
+#include <cuda_utils.h>
+#include <limits.h>
+#include <cub/cub.cuh>
+
+namespace ML {
+namespace SVM {
+
+__global__ void map_to_sorted(const bool *available, int n_rows,
```

Comment:
Please check again binary_op.h or unary_op.h, or functions in linalg folder. One of them should be used instead of this custom kernel.